storage: TestNewTruncateDecision failed under stress [skipped]

SHA: https://github.com/cockroachdb/cockroach/commits/ca1ef4d4f8296b213c0b2b140f16e4a97931e6e7

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNewTruncateDecision PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1368520&tab=buildLog

```
=== RUN   TestNewTruncateDecision
I190701 12:12:45.945806 12152 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190701 12:12:45.946055 12152 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
--- FAIL: TestNewTruncateDecision (0.24s)
    raft_log_queue_test.go:570: truncation should not have occurred, but truncatableIndexes changed from 1 -> 2
    raft_log_queue_test.go:573: truncation should not have occurred, but oldestIndex changed from 116 -> 117

```


Jira issue: CRDB-5625

--- a/pkg/kv/kvserver/raft_log_queue_test.go
+++ b/pkg/kv/kvserver/raft_log_queue_test.go
@@ -24,7 +24,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -450,7 +449,11 @@ func TestNewTruncateDecision(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 38584)
+	// Unquiescing can add spurious empty log entries. Just disable it.
+	testingDisableQuiescence = true
+	defer func() {
+		testingDisableQuiescence = false
+	}()
 
 	ctx := context.Background()
 	stopper := stop.NewStopper()

